### PR TITLE
docs: update trim messages notebook

### PR DESCRIPTION
**Pull request by eyurtsev**
Update trim messages notebook to include common use cases and explain what the desired behavior is

### CODE REVIEW

File: docs/docs/how_to/trim_messages.ipynb [hunk 1]
```diff
@@ -70,73 +114,123 @@
     "    HumanMessage(\"what do you call a speechless parrot\"),\n",
     "]\n",
     "\n",
+    "\n",
     "trim_messages(\n",
     "    messages,\n",
-    "    max_tokens=45,\n",
+    "    # Keep the last <= n_count tokens of the messages.\n",
     "    strategy=\"last\",\n",
+    "    # highlight-start\n",
```
Comment (ccurme) [inline docs/docs/how_to/trim_messages.ipynb:122]: very nit but I was a little confused as to why these particular parameters are highlighted. another option is to just set
```python
token_counter = ChatOpenAI(model="gpt-4o")  # can also pass a custom token counter
max_tokens=45  # adjust to desired history length
```
right above `trim_messages(...)`

**Review by ccurme — COMMENTED** [review]
Guide still features `RunnableWithMessageHistory` at the end. Do we intend to replace that?

File: docs/docs/how_to/trim_messages.ipynb (hunk 1 above)
Comment (eyurtsev) [reply docs/docs/how_to/trim_messages.ipynb:122]: Got it

**Comment by eyurtsev** [comment]
I'll merge as is, but we can update afterwards -- the role of the highlighting is to prevent users from copy pasta without adjusting the parameters appropriately

**Pull request merged**